app/scripts/controllers/network/network-controller.test.js

```diff
@@ -52,147 +55,202 @@ const BLOCK = {
-      getLatestBlockStub = sinon
-        .stub(networkController, '_getLatestBlock')
-        .callsFake(() => Promise.resolve({}));
-      setProviderTypeAndWait = () =>
-        new Promise((resolve) => {
-          networkController.on(NETWORK_EVENTS.NETWORK_DID_CHANGE, () => {
-            resolve();
-          });
-          networkController.setProviderType('mainnet');
-        });
-    });
-
     afterEach(() => {
-      getLatestBlockStub.reset();
-      networkController.destroy();
       nock.cleanAll();
     });
 
     describe('#provider', () => {
       it('provider should be updatable without reassignment', async () => {
-        await networkController.initializeProvider();
-        const providerProxy =
-          networkController.getProviderAndBlockTracker().provider;
-        expect(providerProxy.test).toBeUndefined();
-        providerProxy.setTarget({ test: true });
-        expect(providerProxy.test).toStrictEqual(true);
+        await withController(async ({ controller }) => {
+          setupMockRpcBlockResponses();
+          await controller.initializeProvider();
+          const providerProxy =
+            controller.getProviderAndBlockTracker().provider;
+          expect(providerProxy.test).toBeUndefined();
+
+          providerProxy.setTarget({ test: true });
+
+          expect(providerProxy.test).toStrictEqual(true);
+        });
       });
     });
 
     describe('destroy', () => {
       it('should not throw if called before initialization', async () => {
-        await expect(
-          async () => await networkController.destroy(),
-        ).not.toThrow();
+        const controller = new NetworkController(defaultControllerOptions);
+        await expect(async () => await controller.destroy()).not.toThrow();
       });
 
       it('should stop the block tracker for the current selected network', async () => {
-        nock('http://localhost:8545')
-          .persist()
-          .post(/.*/u)
-          .reply(200, () =>
-            JSON.stringify(constructSuccessfulRpcResponse(BLOCK)),
-          );
-        await networkController.initializeProvider();
-        const { blockTracker } = networkController.getProviderAndBlockTracker();
-        // The block tracker starts running after a listener is attached
-        blockTracker.addListener('latest', () => {
-          // do nothing
-        });
-        expect(blockTracker.isRunning()).toBe(true);
+        await withController(async ({ controller }) => {
+          setupMockRpcBlockResponses();
+          await controller.initializeProvider();
+          const { blockTracker } = controller.getProviderAndBlockTracker();
+          // The block tracker starts running after a listener is attached
+          blockTracker.addListener('latest', () => {
+            // do nothing
+          });
+          expect(blockTracker.isRunning()).toBe(true);
 
-        networkController.destroy();
+          controller.destroy();
```

Comment:
is this already handled by [the `controller.destroy` from inside `withController`](https://github.com/MetaMask/metamask-extension/pull/17250/files#diff-d4e78728faadb83c4ed8c0d5b5b580b026993a8988156b27bf7f41c4b5217af4R87)?

Reply:
ah nvm this is testing that destroy works as specified. disregard.